### DISCUSSION TITLE
Improper rendering of Spine 3D (showing broken instead of a crack)

### DISCUSSION THREAD

**Discussion by ikhizarwork** [google/model-viewer#4469] [Q&A] [answered]
I am using the google's model viewer to show 3D to the users , but i saw some strange issue while i uploaded a Spine injury 3D file 
the spin contains some injury which are mainly a crack , but in the model viewer it is showing broken , which is not true .
so please look up for this issue and try to revert back with the solution asap .
I am attaching some snaps from the Google's model viewer 3D view and some 3rd party 3D viewer app as well , check the difference yourself too .

THE BELOW IMAGE IS OF GOOGLE MODEL VIEWER :- 

IMAGE NO 1 :- <img width="628" alt="broken 3D" src="https://github.com/google/model-viewer/assets/120770287/ce10c09d-d932-48aa-8f90-a71f1ec40943">

IMAGE NO 2 :- 
<img width="699" alt="Broken2" src="https://github.com/google/model-viewer/assets/120770287/d9affe75-1a69-4eca-801a-4311a61ee88f">

THE BELOW IMAGE IS OF 3rd PARTY VIEWER :-

<img width="945" alt="Screenshot 2023-09-20 154123" src="https://github.com/google/model-viewer/assets/120770287/569c0d55-d186-4bd4-b5e7-8ca2080e144c">

IMAGE NO 2 :- 

<img width="945" alt="Screenshot 2023-09-20 154123" src="https://github.com/google/model-viewer/assets/120770287/a058938a-b154-4a92-9218-8a67bf1ac22c">

there is lot of difference betn this 2 viewers , the later image is more correct and better .

**Comment by elalish** [comment, marked as answer]
First, have you checked that the GLB is valid? Khronos has a site for this, and our editor can tell you too. If that's okay, then my best guess is that some triangles are backwards. You can try forcing doublesided to true in our editor to see if that makes a difference: 
<img width="352" alt="image" src="https://github.com/google/model-viewer/assets/1649964/fe7afe2d-d7b3-41f0-907d-47ad4918046f">

**Reply by ikhizarwork** [reply]
i followed your instruction with regarding to checking the validation of the file , and after doing so , i got the response "The Asset contains errors" , after that i followed your next instruction where i opened the editor and set the doublesided to true , and after doing so , i was able to solved the issue . so Thanks alot !

also so just wanted to ask you , since i am using some sort of software to convert the dicom to 3D , so how to prevent such sort of issues for the future , so that the output generated is correct and the end user sees the right 3D . 

hope to see na answer from your end soon :)

**Reply by elalish** [reply]
All it means is that GLB export software you used has bugs. I would recommend filing bug reports on it (they should never produce an invalid GLB, and backwards triangles is also bad practice, but if they can't fix that, they *must* set doublesided to true). Your other viewer also has a bug, since it doesn't respect the doublesided property. Other than that, perhaps find better software?
